api/v1alpha1/deploymenttarget_types.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// DeploymentTargetSpec defines the desired state of DeploymentTarget
+type DeploymentTargetSpec struct {
+	DeploymentTargetClassName DeploymentTargetClassName `json:"deploymentTargetClassName"`
+
+	KubernetesClusterCredentials DeploymentTargetKubernetesClusterCredentials `json:"kubernetesCredentials"`
+
+	ClaimRef string `json:"claimRef"`
```

Comment:
Can the `claimRef` be optional in some cases, eg it should have the `omitEmpty` tag?

Reply:
Good catch. It could be optional